syntax = "proto3";

package pokemon_ou;

service gameserver {
    rpc Captured(Empty) returns (CapturedMessage){}
    rpc Moves(Empty) returns (MovesRecord){}
    rpc Board(Empty) returns (Empty){}
    rpc Connect(ConnectMessage) returns (ConnectResponse){}
    rpc MoveRequest(MoveRequestMessage) returns (Feedback){}
    rpc BoardCheck(CurrentLocation) returns (PossibleMoves){}
}



message Empty {}
message CapturedMessage {
    string status = 1;
}

message MovesRecord {
    repeated string moves = 1;
}

message ConnectMessage {
    string type = 1;
}
message MoveRequestMessage {
    string move = 1;
}
message Feedback {
    string status = 1;
}
message CurrentLocation {
    int32 location = 1;
}
message PossibleMoves {
    repeated int32 moves = 1;
}
message ConnectResponse {
    string status = 1;
    int32 pos = 2;
}